src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -857,6 +849,34 @@ public virtual HtmlString SetAntiforgeryCookieAndHeader()
             return HtmlString.Empty;
         }
 
+        public static bool ShouldRenderAttributeValue(object rawAttributeValue)
+        {
+            if (rawAttributeValue == null)
+            {
+                // Nothing to write
+                return false;
+            }
+
+            // Intentionally using is+cast here for performance reasons. This is more performant than as+bool?
+            // because of boxing.
```

Comment:
(A special case for `rawAttributeValue==null` isn't necessary in `GetStringAttributeValue()`; will already return `null` in that case. And `ShouldRenderAttributeValue()` won't be needed once `GetStringAttributeValue()` handles `rawAttributeValue==false`.)